### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by mcarpenterjr** [Glavin001/atom-beautify#947]
### Description

Installing atom-beautify fails, both through atom and in the command line. Error is the same both ways. 

### Expected Results

Failed output:

```
C:\Windows\system32>npm cache clean -f
npm WARN using --force I sure hope you know what you are doing.

C:\Windows\system32>apm install atom-beautify
Installing atom-beautify to C:\Users\Mark N Carpenter Jr\.atom\packages failed

> spawn-sync@1.0.15 postinstall C:\Users\MARKNC~1\AppData\Local\Temp\apm-install
-dir-116324-5396-19zaics\node_modules\atom-beautify\node_modules\cross-spawn\nod
e_modules\spawn-sync
> node postinstall

Installing native dependencies (this may take up to a minute)

npm WARN engine analytics-node@2.1.0: wanted: {"node":">= 0.12.x"} (current: {"n
ode":"0.10.40","npm":"2.13.3"})
npm WARN engine typescript-formatter@2.1.0: wanted: {"node":">= 4.2.0"} (current
: {"node":"0.10.40","npm":"2.13.3"})
npm WARN deprecated lodash@1.2.1: lodash@<3.0.0 is no longer maintained. Upgrade
 to lodash@^4.0.0.
npm WARN optional dep failed, continuing fsevents@1.0.11
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"}
(current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.
10.40","npm":"2.13.3"})
npm WARN optional dep failed, continuing source-map@0.1.43
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.
10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.
10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.
10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.
10.40","npm":"2.13.3"})
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Users\\Mark N Carpenter Jr\\AppData\\Local\\atom\\app-1.7.2\\
resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Mark N Carpenter Jr\\AppData\\Lo
cal\\atom\\app-1.7.2\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "
--globalconfig" "C:\\Users\\Mark N Carpenter Jr\\.atom\\.apm\\.apmrc" "--usercon
fig" "C:\\Users\\Mark N Carpenter Jr\\.atom\\.apmrc" "install" "C:\\Users\\MARKN
C~1\\AppData\\Local\\Temp\\d-116324-5396-13k1cv0\\package.tgz" "--target=0.36.8"
 "--arch=ia32"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! file C:\Users\Mark N Carpenter Jr\.atom\.apm\amdefine\1.0.0\package\pac
kage.json
npm ERR! code EJSONPARSE

npm ERR! Failed to parse json
npm ERR! Unexpected token '\u0000' at 1:1
npm ERR!

npm ERR! ^
npm ERR! File: C:\Users\Mark N Carpenter Jr\.atom\.apm\amdefine\1.0.0\package\pa
ckage.json
npm ERR! Failed to parse package.json data.
npm ERR! package.json must be actual JSON, not just JavaScript.
npm ERR!
npm ERR! This is not a bug in npm.
npm ERR! Tell the package author to fix their package.json file. JSON.parse

```

### Steps to Reproduce

1. Un-install atom-beautify
2. Attempt to re-install atom-beautify
3. atom-beautify fails on what looks to me on spawn-sync.

My previous issue had been closed in hopes that v0.29.4 would have fixed the issue. I understand this has something to do with a third-party plugin. I find it odd that spawn-sync cannot find the most current version of node, as I have tried with both of the most current releases, V4.x.x and V5.x.x both the 32 bit and 64 bit versions as well.

### Debug

No debug info as atom-beautify does not install.

#### How to create `debug.md` Gist

Note that this will include a copy of your code.
If your code is private, please create a different sample of code that reproduces the problem.

1. In the Atom [command-palette](https://atom.io/packages/command-palette), 
search for and run the command `Atom Beautify: Help Debug Editor`. 
The debugging results will be copied to your clipboard.  
2. Create a new Gist at https://gist.github.com/  
3. Create a file in your new Gist called `debug.md`.  
4. Paste your debugging results from Atom beautify into `debug.md` file in your Gist.  
5. Add a link to your Gist in your new Issue.

# Checklist

- [X] I have tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [X ] Searched for existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues 
so I know this is not a duplicate issue
- [ ] Generated debugging information and added link for `debug.md` Gist to this issue

**Comment by Glavin001** [comment]
> npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! file C:\Users\Mark N Carpenter Jr\.atom\.apm\amdefine\1.0.0\package\pac
kage.json
npm ERR! code EJSONPARSE
>
npm ERR! Failed to parse json
npm ERR! Unexpected token '\u0000' at 1:1
npm ERR!
>
npm ERR! ^
npm ERR! File: C:\Users\Mark N Carpenter Jr\.atom\.apm\amdefine\1.0.0\package\pa
ckage.json
npm ERR! Failed to parse package.json data.

From the logs I am not sure what is causing this issue. It looks like it may be related to `apm` and/or `amdefine`. Maybe someone over at https://github.com/atom/apm would be better qualified to help out and provide some advice.

**Comment by mcarpenterjr** [comment]
Cool, thank you.

**Comment by mcarpenterjr** [comment]
So I am closing this issue, because I have reached a resolution. I ultimately did a clean install of atom, github desktop, node and python. to rule out all major suspects. Ultimately though, I can conclude in the windows user temp folder ``` %AppData%\Local\Temp\``` dirty installs and upgrades for what ever reason are being left behind, simple restarts won't clear them. I deleted all folders associated with npm, apm and atom before doing a fresh install. This cleared the issue, took a couple trial and errors to make sure old packages were not left behind. 
Also clearing ``` C:\Users\Your_User_Name\.atom ``` may help. Although I believe this is for current packages and configurations.

I did also find that most of these install errors of this nature sometimes relate to deep file trees created when installing packages and their dependencies, windows does not like deep file trees.
